ovve_ui/utils/comms_link.py

```diff
@@ -0,0 +1,278 @@
+import json
+import random
+import time
+from threading import Thread
+import serial
+import sys
+from time import sleep
+import binascii
+import crc16
+import codecs
+
+from utils.comms_adapter import CommsAdapter
+from utils.params import Params
+from utils.settings import Settings
+
+
+class CommsLink():
+    def __init__(self, comms_adapter: CommsAdapter) -> None:
+        self.comms_adapter = comms_adapter
+        self.comms_adapter.set_comms_callback(self.update_settings)
+        self.done = False
+        self.settings = Settings()
+        self.seqnum = 0
+        self.packet_version = 1
+        self.BAUD = 38400
+        self.PORT = "/dev/ttyACM0"
+        self.SER_TIMEOUT = 0.055
+        self.ser = 0
+        self.crcFailCnt = 0
+        self.init_serial()
+        
+
+    def update_settings(self, settings_dict: dict) -> None:
+        self.settings.from_dict(settings_dict)
```

Comment:
I think we need a mutex here where we write the settings from the UI thread, and where we read it on the serial thread.

Reply:
Added by Brian